packages/saber/vue-renderer/lib/index.js

```diff
@@ -403,18 +403,18 @@ class VueRenderer {
     })
 
     server.get('/_saber/visit-page', async (req, res) => {
-      const pathname = removeTrailingSlash(decodeURI(req.query.route))
+      const pathname = normalizeRoutePath(req.query.route)
       log.info(`Navigating to ${pathname}`)
       res.end()
 
       if (this.builtRoutes.has(pathname)) {
-        hotMiddleware.publish({ action: 'router:push', route: pathname })
+        hotMiddleware.publish({ action: 'router:push', route: req.query.route })
```

Comment:
We need to ensure there's no trailing hash in `req.query.route`

Reply:
Got it. 

https://github.com/saberland/saber/pull/270/commits/52ac92448940b3a9dd9dd648d35a23675c673238

Reply:
oops, it's a typo, I mean `slash` 😆

Reply:
Oh. 

But also I kinda liked how you refactored it. I definitely learnt something after just waking up 😉